polyfactory/utils/helpers.py

```diff
@@ -29,7 +31,7 @@ def unwrap_union(annotation: Any) -> Any:
     :returns: A type annotation
     """
     while is_union(annotation):
-        annotation = get_args(annotation)[0]
+        annotation = choice(get_args(annotation))
```

Comment:
This can't be used. You're modifying a helper that is used in multiple places. 

You can add a localised change and use `factory.__random__.choice`

Reply:
Yes, I've modified the helper for Union. The question is shouldn't that helper always choose one of all expected Union types? I think that taking always the first element is wrong. Moreover, in the previous library "pydantic-factory" that also didn't work like that - Previously in that situation randomly one of the allowed types was used.

Reply:
I'm aware of what was in V1, this was changed intentionally in this helper because you cannot move the factory random into it easily. 

That's why I say you can add a different fix instead of using the helper, but you cannot modify tue helper. The results need to be deterministic on a random seed that is tied to the factory.

Reply:
The issue is that this helper is a general typing utility, not only used to generate values. If you call the helper because you want to know the inner type of a union of e.g `str | list[str]`, it can't *sometimes* return `str` and sometimes `list[str]`. The randomisation should happen at the point where the values are being consumed.

Reply:
@provinzkraut thanks for the detailed explanation :)

Reply:
Thanks 👍, he always explains better than me 😂. 

The point is that you can use random.choice, because the results have to deterministic when seeding the factory random. So we have to use `factory.__random__.choice`. This means you can't use the helper, rather you should create a different solution that allows this, or update the helper to pass random as an arg/kwarg. Bit that might get involved and harder. 

I hope this is clearer